### PR TITLE
Feat: Adding provenance publish config

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What this PR does / why we need it**:
Linking npm package builds better to source code and CI builds using sigstore.

More context in: https://docs.npmjs.com/generating-provenance-statements

I don't think we have to force publishing new versions just because of this change. I would rather just wait for regular releases to happen to those packages after this change is merged.

### CODE REVIEW

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -186,6 +186,9 @@ jobs:
       NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
       NX_BRANCH: ${{ github.event.number || github.ref_name }}
       NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+    permissions:
```
Comment (tolzhabayev) [inline .github/workflows/ci.yml:189]: @jackw I've added this because I assume it will be required. fyi

Comment (jackw) [reply .github/workflows/ci.yml:189]: Looking at [the docs](https://docs.npmjs.com/generating-provenance-statements#example-github-actions-workflow) we do need to set the `id-token`. However are we using this github token or the `github-app-token` further down?

Comment (tolzhabayev) [reply .github/workflows/ci.yml:189]: My assumption is that they are a bit different and it needs the specific id-token. I guess we will see once we run it :)

**Pull request merged**